### PR TITLE
tests: add content sharing interface spread test

### PR DESCRIPTION
**Pull request by fgimenez**
Currently checks for the slot-plug autoconnection, reconnection and fstab files creation.

### CODE REVIEW

File: tests/lib/snaps/content-slot/meta/snap.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: content-slot
+version: 1.0
+slots:
+    shared-content-slot:
+        interface: content
+        content: mylib
+        read:
+            - $SNAP_DATA
```
Comment (mvo5) [inline tests/lib/snaps/content-slot/meta/snap.yaml]: I'm not sure this works, could you please try just `/` here? The path is always relative to $SNAP_DATA here.

File: tests/main/interfaces-content/task.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
+
+    echo "And fstab files are created"
+    [ $(find /var/lib/snapd/mount -type f -name "*.fstab" | wc -l) -gt 0 ]
+
+    echo "============================================"
+
+    echo "When the plug is disconnected"
+    snap disconnect content-plug:shared-content-plug content-slot:shared-content-slot
+    snap interfaces | grep -Pzq "$DISCONNECTED_PATTERN"
```
Comment (mvo5) [inline tests/main/interfaces-content/task.yaml]: We could check here that the fstab file vanishes maybe?

File: tests/main/interfaces-content/task.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    echo "Then the plug can be connected again"
+    snap connect content-plug:shared-content-plug content-slot:shared-content-slot
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
```
Comment (mvo5) [inline tests/main/interfaces-content/task.yaml]: And that is reappears here?

**Comment by mvo5** [comment]
Thanks for working on this! Looks very nice, some small suggestions plus it actually fails :) I suspect the read: $SNAP might cause the failure.

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: tests/lib/snaps/content-slot/meta/snap.yaml (hunk 1 above)
Comment (fgimenez) [inline tests/lib/snaps/content-slot/meta/snap.yaml]: Sure, changed to `/`

**Comment by fgimenez** [comment]
@mvo5 thx! comments addressed, pls have another look

**Comment by niemeyer** [comment]
Test is currently broken.

**Comment by fgimenez** [comment]
@niemeyer yes, the fstab files are not being created. I'm not sure if the test snaps are properly defined, waiting for @mvo5 to have a look

File: tests/lib/snaps/content-slot/meta/snap.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: content-slot
+version: 1.0
+slots:
+    shared-content-slot:
+        interface: content
+        content: mylib
+        read:
```
Comment (zyga) [inline tests/lib/snaps/content-slot/meta/snap.yaml]: Given this is always relative to the snap name I wonder if there's any test that we can do for `write`.

Comment (fgimenez) [inline tests/lib/snaps/content-slot/meta/snap.yaml]: hey @zyga, any suggestion is more than welcome :blush:

File: overlord/ifacestate/helpers.go [hunk 5]
```diff
@@ -26,6 +26,10 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/backends"
 	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/dbus"
```
Comment (mvo5) [inline overlord/ifacestate/helpers.go]: This part of the diff is no longer needed (the entire helpers.go file change), this got refactored. Just appyl:
```
diff --git a/overlord/ifacestate/helpers.go b/overlord/ifacestate/helpers.go
index cd0c7d1..d0690f2 100644
--- a/overlord/ifacestate/helpers.go
+++ b/overlord/ifacestate/helpers.go
@@ -26,10 +26,6 @@ import (
        "github.com/snapcore/snapd/interfaces"
        "github.com/snapcore/snapd/interfaces/backends"
        "github.com/snapcore/snapd/interfaces/builtin"
-       "github.com/snapcore/snapd/interfaces/dbus"
-       "github.com/snapcore/snapd/interfaces/mount"
-       "github.com/snapcore/snapd/interfaces/seccomp"
-       "github.com/snapcore/snapd/interfaces/udev"
        "github.com/snapcore/snapd/logger"
        "github.com/snapcore/snapd/overlord/snapstate"
        "github.com/snapcore/snapd/overlord/state"
@@ -216,7 +212,3 @@ func getConns(st *state.State) (map[string]connState, error) {
 func setConns(st *state.State, conns map[string]connState) {
        st.Set("conns", conns)
 }
-
-var securityBackends = []interfaces.SecurityBackend{
-       &seccomp.Backend{}, &dbus.Backend{}, &udev.Backend{}, &mount.Backend{},
-}
```

**Comment by mvo5** [comment]
Looks fine and can go in, needs to undo the helper.go change first though as this got fixed somewhere else now.

**Comment by fgimenez** [comment]
@mvo5 done, thanks! :)

**Pull request merged**